pkg/agent/baker_test.go

```diff
@@ -142,8 +173,14 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
 			configUpdator(config)
 		}
 
+		configCustomDataClone, err := deepcopy.Anything(config)
```

Comment:
this avoids mutation

Reply:
why do we need two deep copies of the same thing? 
configCseClone, err := deepcopy.Anything(config) and configCustomDataClone, err := deepcopy.Anything(config)

Reply:
to avoid mutation of the original config -- this is what happened with the dynamic config dir bug.

GetNodeBootstrappingPayload mutates the input so it's not the same as what gets passed to GetNodeBootstrappingCmd which causes bugs. agentbaker as a service uncovered this.

will comment.

Reply:
Hmm gotcha, but why are we deep copying the same thing twice? In the end we have configCustomDataClone and configCseClone - different names, but wouldn't the result be the same thing? Both are created by deepcopy.Anything(config) using the same config variable.

Reply:
I couldn't decide which one to use the original thing for, so I renamed both :P

Reply:
I can strictly do one fewer copy but I don't mind being explicit for the test logic with a big WARNING

Reply:
Two copies are prefectly fine, I was just confused by the naming i guess 😄